locator/topology.cc

```diff
@@ -438,24 +438,29 @@ const node* topology::find_node(node::idx_type idx) const noexcept {
     return _nodes.at(idx).get();
 }
 
-const node* topology::add_or_update_endpoint(inet_address ep, std::optional<host_id> opt_id, std::optional<endpoint_dc_rack> opt_dr, std::optional<node::state> opt_st, std::optional<shard_id> shard_count)
+const node* topology::add_or_update_endpoint(std::optional<inet_address> opt_ep, std::optional<host_id> opt_id, std::optional<endpoint_dc_rack> opt_dr, std::optional<node::state> opt_st, std::optional<shard_id> shard_count)
```

Comment:
Can we make `opt_id` non-optional (by type, not runtime check) at the end?

Since we turned `ep` into `opt_ep`.

Reply:
Yes, already did in the latest PR version.

Reply:
I don't think you did. In the full diff, you can see `opt_id` being optional still.

Reply:
Turned out I somehow lost this patch((( Will redo it again